File: TP1/TP1 python/Tomas/Ejericicio 2/Ticket.py
from datetime import datetime


class Ticket:
    def __init__(self, nroVenta):
        self.nroVenta = nroVenta
        self.fecha = datetime.now()
        self.producto = []
    
    def agregarProducto(self, producto):
        self.producto.append(producto)
    
    def calcularTotal(self):
        total = 0.0
        for produ in self.producto:
            total += produ.getPrecio()
        return total
    
    def imprimir(self):
        print("Nro de venta: {}          Fecha: {}".format(self.nroVenta, self.fecha))
        print("Articulo       Dosis          Precio")
        for produ in self.producto:
            produ.imprimir()
        print("                         Total:", self.calcularTotal())
        
    def getNroVenta(self):
        return self.nroVenta
    
    def getgramaje(self):
        return self.gramaje

File: Practica Parcial/Cesar/Parcial 2023/Parcial 23/src/Ticket.java

import java.time.LocalDate;
import java.util.ArrayList;

public class Ticket {
    private Integer nroventa;
    private LocalDate fecha; 
    
    private ArrayList<Producto> listaprod = new ArrayList<>();
    
    public Ticket(Integer nroventa, LocalDate fecha) {
        this.nroventa = nroventa;
    }
    public void agregarProducto(Producto produ){
        listaprod.add(produ);
    }
    public void ImprimirProductos(){
        System.out.println("Articulo " +"            "+ " Dosis"+"            " + "Precio"+"\n");
        for(Producto p : listaprod){
            System.out.println(p);
        }
    }
    public String Total(){
        Integer Ptotal = 0;
        for (Producto p : listaprod){
            Ptotal = Ptotal + p.getPrecio();
        }
        return "Total = "+ Ptotal;
    }
    public String tostring(){
        return "Numero de venta : "+nroventa+"                        "+"Fecha: "+fecha+"\n"; 
    }
}
